README.md

```diff
@@ -99,6 +99,12 @@ To create a bridge and two (or `count` at your option) tap interfaces:
 To contribute something to RIOT, please refer to our
 [contributing document](CONTRIBUTING.md).
 
+## QUESTIONS
```

Comment:
i would just replace Questions by Stackoverflow to be consistent with the other captions..

we still encourage questions via mailing lists ;).

Reply:
> we still encourage questions via mailing lists ;).

I don't think mailing lists are suitable for asking questions. They are not really searchable (leading to duplicate questions), and you have no ability to just follow specific discussion threads. The amount of questions on the mailing list also seems *really* low compared to the activity here on GitLab (same for StackOverflow by the way).

Additionally, there [are][1] [several][2] [unanswered][3] [questions][4] in the last few months.
Not sure how StackOverflow compares, though.

[1]: https://lists.riot-os.org/pipermail/users/2020-August/001595.html
[2]: https://lists.riot-os.org/pipermail/users/2020-July/001573.html
[3]: https://lists.riot-os.org/pipermail/users/2020-June/001558.html
[4]: https://lists.riot-os.org/pipermail/users/2020-May/001549.html

Reply:
in any case, we don't want to dictate where people can raise questions.

all of the unanswered questions you refer to have been raised on the users mailing list. there is a tendency that most RIOTers are active on devel. but this ia another topic.

Reply:
> in any case, we don't want to dictate where people can raise questions.

Agreed, but if someone has a question they should be pointed to a *preferred* medium. From a user perspective it's useful to have 1) a question section in the README, which 2) directs to an accessible way of asking the question.

This is not to say that StackOverflow *is* that preferred medium. But I think the section should probably be kept, possibly with links to other official communication channels. For example:

```
# QUESTIONS

Questions can be asked on `<x>`.
It is also possible to ask questions via the [`riot-os` tag on Stackoverflow](https://stackoverflow.com/questions/tagged/riot-os),
on [the RIOT OS users mailing list](https://lists.riot-os.org/mailman/listinfo/users),
or on [Matrix](https://matrix.to/#/#riot-os:matrix.org).
```

With the list correctly duplicated, of course :)

Reply:
I think, as a user, that StackOverflow is not a valid medium. Be using a tag on StackOverflow you move Riot-OS community questions to a different community. Where I can predict that in no-time one of the upvoted answers would be: "I just used MBED/Zephir and it worked fine for me". And especially for development questions like "I wrote a driver for device X, do I want to implement that like A or B", StackOverflow is not the way to go.

However, an alternative for mailing lists would be very welcome though. From time to time I like to play with Riot-OS, but I have no experience with mailing lists. And until a couple of weeks ago, I knew mailing lists from obscure computer questions. Where internet search would find you an archived mailing list from 2007. So I was amazed that mailing lists are still a thing today.
With my current knowledge about mailing lists I still do not like the effort and overhead of managing a mailing list subscription. Especially as I have a question once every two months.

The third option, Matrix, is also not very welcome. If I want to ask a new question on that channel once every two month. I am kicked out of the channel after 30 days of inactivity. And similar to the mailing list, it is not feasible for me to keep track of all the backlog.

A StackOverflow-like forum inside the RIOT-community would be my prefered channel. It is searchable, but above all readable without having to join something. Furthermore, you can more easily improve/edit solutions to questions when over the years RIOT changes and maybe the optimal solution aswell.

Reply:
I think that Achilleshiel has a point about Stackoverflow, the quality of answers is not always as good.
However, as someone who sometimes uses riot for a few hobby projects, I would definitely prefer it over the current solution.

I know that one can use the mailing list to ask a question.
However, something that would allow me to search in old questions, and maybe even see accepted answers, would be way faster for me. And is faster, less effort and probably more complete than a single mailing list answer.

So even if StackOverflow is not the ideal platform for these discussions, it is one that is apparently already in use and would require less effort then a custom option.

Other solutions would in my opinion be:
  - A searchable/tagged archive of all mailing list questions.
  - A BBS to remain within the technological era of the mailing lists 😉 (or a more modern forum).
  - Keep bothering your local RIOT expert with questions. (if you have one).

Reply:
> > we still encourage questions via mailing lists ;).
> 
> I don't think mailing lists are suitable for asking questions. They are not really searchable (leading to duplicate questions), and you have no ability to just follow specific discussion threads. The amount of questions on the mailing list also seems _really_ low compared to the activity here on GitLab (same for StackOverflow by the way).
> 
> Additionally, there [are](https://lists.riot-os.org/pipermail/users/2020-August/001595.html) [several](https://lists.riot-os.org/pipermail/users/2020-July/001573.html) [unanswered](https://lists.riot-os.org/pipermail/users/2020-June/001558.html) [questions](https://lists.riot-os.org/pipermail/users/2020-May/001549.html) in the last few months.
> Not sure how StackOverflow compares, though.

I completely agree, this is the main reason why I pushed it on Stackoverflow. There are also some unanswered questions, but it is more simple to keep track of them, I think. I also think there is a huge benefit in marketing on put all questions on Stackoverflow.

Reply:
> I think, as a user, that StackOverflow is not a valid medium. Be using a tag on StackOverflow you move Riot-OS community questions to a different community. Where I can predict that in no-time one of the upvoted answers would be: "I just used MBED/Zephir and it worked fine for me". And especially for development questions like "I wrote a driver for device X, do I want to implement that like A or B", StackOverflow is not the way to go.

This is subjective and will result in a lot of downvotes on stackoverflow. Maybe it is even against the guidlines. There needs to be an objective, understandable response on a question. So, you can say: Contiki is better in this and that and RIOT OS is better than this. I wouldn't be so overprotective on this topic. Just create a great product and the rest will just come. Overprotective behavior will eventually lead in the opposite, I think. You are not getting a lot of attention and therefore not a lot of new users. If there is valid critic we can always work on it to change it. And also, everyone is allowed to respond to these questions.

Reply:
> You can also add new answers to old questions. No big deal. I already did it and even resulted in one of my most successful answers.
Offcourse, and that is a good thing. However, your answer is still the second upvoted and not the accepted answer. And especially if an originally upvoted and accepted answers has become a security issue or really bad practice you want to remove that. Furthermore, often I cannot determine if a newer answer is better, or just easier. If I would understand the subject completely, I wouldn't search on StackOverflow.

Reply:
> > You can also add new answers to old questions. No big deal. I already did it and even resulted in one of my most successful answers.
> 
> Offcourse, and that is a good thing. However, your answer is still the second upvoted and not the accepted answer. And especially if an originally upvoted and accepted answers has become a security issue or really bad practice you want to remove that. Furthermore, often I cannot determine if a newer answer is better, or just easier. If I would understand the subject completely, I wouldn't search on StackOverflow.

The most upvoted answer is the most recent and a correct answer. This is what some people prefer in some scenarios. Not all use-cases fit into my answered question. The accepted answer was, at the time, correct. There was not equivalent to LIKE. The upvotes should show you that people are aware of developments and read the newer and helpful answers. Your mentioned security issues etc. are very rare occasions. An OS should also provide backwards compatibility. Removing APIs should happen rarely. 

Overall can we continue the way we currently handle things, but Contiki, FreeRTOS and ZephyrOS will be more successful with it. Stackoverflow is also not supposed to be for core development questions.  It's more for users of RIOT.

I would also like to have a forum on riot-os.org. Would be nice.

The change by @silkeh looks fine to me.

Reply:
still, it is unclear what <x> in the proposed text is. i tend to prefer a bullet list.

and i would not remove the users mailing list from the Section MAILING LISTS because readers who explicitly jump to the section might miss this channel otherwise.

Reply:
> I would also like to have a forum on riot-os.org. Would be nice.

Agreed, a Flarum or Discourse would probably be nice. The latter can also (somewhat) be used like mailing lists. However, seeing the low traffic on StackExchange and the mailing list compared to the [type: question tag](https://github.com/RIOT-OS/RIOT/issues?q=is%3Aissue+label%3A%22Type%3A+question%22+) makes me wonder if questions shouldn't just be asked and answered here on GitHub.

> still, it is unclear what in the proposed text is.

The proposal depends on what the *preferred* medium is for asking questions. I'm not sure what that is, and that's not something *I* can decide :slightly_smiling_face: 

>  i tend to prefer a bullet list.

For an overview of all communication channels I would agree. However, I think this section should be more of a guide, not an overview. I would also expect the overview to be located on the main website or on the wiki (which can be linked in this section), not somewhere in the README. Except for some badges at the top, of course.

> and i would not remove the users mailing list from the Section MAILING LISTS because readers who explicitly jump to the section might miss this channel otherwise.

Agreed, if there is a list it should be complete. However, I have no idea what the purpose of the section is: there is no description of the section or of the mailing lists itself. Additionally, the bullet points make no sense.